src/components/map/SplitView.js

```diff
@@ -7,104 +7,96 @@ import Layer from './layers/Layer';
         }
-    }
-
-    render() {
-        const {
-            isPlugin,
-            basemap,
-            layer,
-            feature,
-            openContextMenu,
-        } = this.props;
-        const { isFullscreen } = this.state;
+        return () => {
+            if (map && isPlugin) {
+                map.off('fullscreenchange', onFullscreenChange);
+            }
+        };
+    }, [map, isPlugin, onFullscreenChange]);
 
-        const { id, periods = [] } = layer;
-
-        return (
-            <div
-                ref={node => (this.node = node)}
-                className={cx('dhis2-map-split-view', styles.splitView)}
-            >
-                {periods.map((period, index) => (
-                    <MapItem
-                        key={period.id}
-                        index={index}
-                        count={periods.length}
-                        layerId={id}
-                        onAdd={this.onMapAdd}
-                        onRemove={this.onMapRemove}
-                        setMapControls={this.setMapControls}
-                        isPlugin={isPlugin}
-                        isFullscreen={isFullscreen}
-                    >
-                        <Layer index={0} {...basemap} />
-                        <ThematicLayer
-                            index={1}
-                            period={period}
-                            feature={feature}
-                            {...layer}
-                            openContextMenu={openContextMenu}
-                        />
-                        <PeriodName period={period.name} />
-                    </MapItem>
-                ))}
-            </div>
-        );
-    }
+    useEffect(() => {
+        // From map plugin resize method
+        setShowFullscreen(isFullscreen);
+    }, [isFullscreen]);
 
-    // Called from map child
-    setMapControls = map => {
-        const controls = {};
+    const { id, periods = [] } = layer;
 
-        this.props.controls.forEach(control => {
-            map.addControl(control);
-            controls[control.type] = map.getControlContainer(control.type);
-        });
+    return (
+        <div
+            ref={containerRef}
+            className={cx('dhis2-map-split-view', styles.splitView)}
+        >
+            {periods.map((period, index) => (
+                <MapItem
+                    key={period.id}
+                    index={index}
+                    count={periods.length}
+                    layerId={id}
+                    setMapControls={setMap}
+                    isPlugin={isPlugin}
+                    isFullscreen={showFullscreen}
+                >
+                    <Layer index={0} {...basemap} />
+                    <ThematicLayer
+                        index={1}
+                        period={period}
+                        feature={feature}
+                        {...layer}
+                        openContextMenu={openContextMenu}
```

Comment:
feature and openContextMenu are kind of "passthru props" and I'm wondering if it makes more sense to just pass them as {...rest} to ThematicLayer.